WordPress/gutenberg
# Full Site Editing: Add a Site Logo block

Props @pablinos who wrote most of the code.

## Description
This adds a Site Logo block to the Full Site Editing experiment.

This uses some filters to non-destructively override the theme mod value when the experiment is enabled and the site logo setting has a value.

## How has this been tested?
- To test, enable the Full Site Editing experiment
- Add a template (e.g. https://github.com/WordPress/gutenberg/pull/18554)
- Add a Site Logo block to the template
- Add a Site Logo block to some content
- Try changing the alignment in the block toolbar
- Try modifying the width in the sidebar
- Try adding alt text in the media library
- Load the front end of the site and check that you can see the Site Logo

## Screenshots
<img width="638" alt="Screenshot 2019-11-28 at 17 18 08" src="https://user-images.githubusercontent.com/275961/69824301-42044780-1203-11ea-9724-cdb431102836.png">
<img width="422" alt="Screenshot 2019-11-28 at 17 17 29" src="https://user-images.githubusercontent.com/275961/69824302-42044780-1203-11ea-9338-c4ea47386405.png">

## Known issues
If a non-admin triese to use this block they won't see anything.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
